src/lib/pages/validator-details/components/tables/ProposedBlocksTable.tsx

```diff
@@ -55,72 +136,66 @@ export const ProposedBlocksTable = ({
   );
 
   if (isLoading) return <Loading />;
-  if (error) return <ErrorFetching dataName="blocks" />;
-  if (!data?.total)
+  if (error || !data) return <ErrorFetching dataName="blocks" />;
+  if (!data.total)
     return (
-      <EmptyState
-        imageVariant="empty"
-        message="This validator never proposed any blocks."
-        withBorder
-      />
+      <>
+        <TableTitle
+          title="Proposed Blocks"
+          count={0}
+          helperText={
+            onViewMore
+              ? ""
+              : "Display the proposed blocks by this validator within the last 30 days"
+          }
+          mb={0}
+        />
+        <EmptyState
+          imageVariant={onViewMore ? undefined : "empty"}
+          message="This validator never proposed any blocks."
+          withBorder
+        />
+      </>
     );
```

Comment:
- `error` is not needed
- let's create a file for the body component
- move all these into the body component and pass `data` and `isLoading` to the body component
- remove table title

Reply:
according to the design, we still need table title even it in empty state ya?